Common/triggerSel.py

```diff
@@ -0,0 +1,105 @@
+import ROOT
+import Common.BaselineSelection as Baseline
+syst_names = {"_Central":Baseline.ana_reco_object_collections, "_TauESUp":["Tau", "MET"], "_TauESDown":["Tau", "MET"]}
+
+
+def ApplyTriggers(df, yaml_dict= None, isData = False):
+    total_objects_matched = []
+    all_or_strings = []
+    total_or_paths = []
+    dict_legtypes = {"Electron":"Leg::e", "Muon":"Leg::mu", "Tau":"Leg::tau"}
+    if yaml_dict is None:
+        return df
+    for path in yaml_dict:
+        path_dict = yaml_dict[path]
+        #channel_or_string = '( '
+        #channel_or_string += ' || '.join(f"httCand.channel() == Channel::{ch} " for ch in path_dict['channels'])
+        #channel_or_string += ' )'
+        or_paths = '( '
+        if('path' in path_dict):
+            or_paths += " || ".join(path for path in path_dict['path'] )
+        else:
+            str_data = 'data' if isData else 'MC'
+            or_paths += " || ".join(path for path in path_dict[f"path_{str_data}"] )
+        or_paths += ' ) '
+
+        leg_dict = path_dict['legs']
+        k=0
+        for leg_tuple in leg_dict:
+            k+=1
+            # define offline cuts
+            leg_dict_offline= leg_tuple["offline_obj"]
+            var_name_offline = f"""{leg_dict_offline["type"]}_offlineCut_{k}_{path}"""
+            type_name_offline = leg_dict_offline["type"]
+            df = df.Define(f"{var_name_offline}", f"""{leg_dict_offline["cut"]}""")
+            if leg_tuple['doMatching'] == False:
+                    if(leg_dict_offline['type']!='MET'):
+                        var_name_offline = f"""{leg_dict_offline['type']}_idx[{var_name_offline}].size()>=0"""
+                    total_or_paths.append(f"""({or_paths} &&  {var_name_offline})""")
+                    continue
+            # require that isLeg
+            df = df.Define(f"{type_name_offline}_{var_name_offline}_sel", f"""httCand.isLeg({type_name_offline}_idx, {dict_legtypes[type_name_offline]}) && ({var_name_offline})""")
+            df = df.Define(f"isHttLegAndPassOfflineSel_{var_name_offline}_{type_name_offline}", f"{var_name_offline} && {type_name_offline}_{var_name_offline}_sel")
+            leg_dict_online= leg_tuple["online_obj"]
+            var_name_online =  f"""{leg_dict_offline["type"]}_onlineCut_{k}_{path}"""
+            df = df.Define(f"""{var_name_online}""",f"""{leg_dict_online["cut"]}""")
+            matching_var = f"""{leg_dict_offline["type"]}_Matching_{k}_{path}"""
+            # find matching online <-> offline
+            df = df.Define(f"{matching_var}", f"""FindMatchingOnlineIndices(isHttLegAndPassOfflineSel_{var_name_offline}_{type_name_offline}, {var_name_online},
+                                           TrigObj_eta, TrigObj_phi, {leg_dict_offline["type"]}_eta,{leg_dict_offline["type"]}_phi, 0.4 )""")
+
+            total_objects_matched.append([matching_var])
+        if(f"""({or_paths} &&  {var_name_offline})""" in total_or_paths):
+            continue
```

Comment:
why this is needed?

Reply:
because otherwise it calculates the `hasHttCandCorrespondance` also when selecting the other paths, the alternative is to require again that the doMatching is false

Reply:
what other paths? `hasHttCandCorrespondance` should be called once for each "path block". If matching is not required for the given leg, it shouldn't be added to `total_objects_matched` vector, so should be transparent.